### PR TITLE
Add a Basic Container association

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: lib/active_fedora/associations.rb [hunk 1]
```diff
@@ -104,6 +106,26 @@ def association_instance_set(name, association)
...
       module ClassMethods
+        # This method is used to declare this resource acts like an LDP BasicContainer
+        #
+        # @param [Hash] options
+        # @option options [String] :class_name ('ActiveFedora::File') The name of the
+        #   class that will represent the contained resources
+        #
+        # example:
+        #   class FooHistory < ActiveFedora::Base
+        #     directly_contains :files, has_member_relation:
```
Comment (cbeer) [inline lib/active_fedora/associations.rb]: Is this comment accurate for `is_a_container`?

File: lib/active_fedora/associations/contains_association.rb [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
+      end
+
       def reader
         @records ||= ContainerProxy.new(self)
       end
 
       def include?(other)
         if loaded?
           target.include?(other)
+        # TODO: fix this
```
Comment (cbeer) [inline lib/active_fedora/associations/contains_association.rb]: What fixing needs to happen?

Comment (jcoyne) [inline lib/active_fedora/associations/contains_association.rb]: Eh, just cruft.

File: lib/active_fedora/associations.rb (hunk 1 above)
Comment (jcoyne) [inline lib/active_fedora/associations.rb]: Updated. Thanks.

File: lib/active_fedora/associations/basic_contains_association.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      def insert_record(record, force = true, validate = true)
+        record.base_path_for_resource = owner.uri.to_s
+        super
+      end
+
+      def add_to_target(record, skip_callbacks = false)
+        record.base_path_for_resource = owner.uri.to_s
+        super
+      end
+    end
```
Comment (jcoyne) [inline lib/active_fedora/associations/basic_contains_association.rb]: This is missing `destroy_all`

Comment (jcoyne) [inline lib/active_fedora/associations/basic_contains_association.rb]: Added to the superclass.

**Pull request merged**